autoload/youcompleteme.vim

```diff
@@ -505,6 +508,8 @@ function! s:OnInsertEnter()
     return
   endif
 
+  call s:SetOmnicompleteFunc()
```

Comment:
But with that if after a restart of the ycmd server a semantic completer flip its status we could not detect it. I have in mind a more complex approach but I don't know if it would be worth it.